astropy/visualization/mpl_style.py

```diff
@@ -75,11 +70,11 @@
                '#188487',   # turquoise
                '#E24A33']   # orange
 
-if MATPLOTLIB_GE_1_5:
+try:
```

Comment:
I don't see the point to even check matplotlib minversion here but maybe @astrofrog or @Cadair can confirm.

Reply:
Or perhaps do the `try/except` if `cycler` is a required dependency?

Reply:
@mhvk , can you please clarify? I already am doing a `try/except`.

Reply:
a lot of the minversions are checked in `__init__.py`, maybe mpl could be added there, too?

Reply:
Just meant that I didn't think the `try/except` was necessary; if `cycler` is a dependency, shouldn't we just be able to count on it being present?

Reply:
@mhvk , `cycler` is required at installation for `matplotlib`. So should I go ahead and just assume it is always there? https://github.com/matplotlib/matplotlib/blob/8d2e360feb7e1a8a9e757c49ae6ea063a7cb2cf4/setup.py#L289-L290

@bsipocz , yeah, good point. Will add that in the next round of changes.

Reply:
I feel this was a remnant from the era when it became a dependency as wasn't installed properly. With 3.0+ I would be surprised if the issue is still around (and if it is, then it should be an upstream issue for mpl to sort out their metadata)

Reply:
Yes, I think we should assume correct installations unless forced otherwise!

Reply:
I think it's probably fine to assume that cycler is installed if MPL is installed

Reply:
If it isn't, 3.x already fails on importing matplotlib or pyplot.

Reply:
Good points, all. Removed try/except check.

Reply:
Well... Nope, not gonna work. Have to add try/except back. See https://app.circleci.com/pipelines/github/astropy/astropy/2939/workflows/a2624ad2-7d6e-47de-b3b8-2efe70241eb0/jobs/80610 and https://travis-ci.com/github/astropy/astropy/jobs/408037544

Reply:
oh, this has to be a big bug in mpl, surely

Reply:
Maybe not, maybe it is just how the tests are being collected. 

```
_ ERROR collecting .tox/py37-test/lib/python3.7/site-packages/astropy/visualization/tests/test_time.py _
../../.tox/py37-test/lib/python3.7/site-packages/pytest_doctestplus/plugin.py:184: in collect
    module = self.fspath.pyimport()
../../.tox/py37-test/lib/python3.7/site-packages/py/_path/local.py:704: in pyimport
    __import__(modname)
../../.tox/py37-test/lib/python3.7/site-packages/astropy/visualization/__init__.py:6: in <module>
    from .mpl_style import *
../../.tox/py37-test/lib/python3.7/site-packages/astropy/visualization/mpl_style.py:8: in <module>
    from cycler import cycler
E   ModuleNotFoundError: No module named 'cycler'
```

Reply:
You could always add it here: https://github.com/astropy/astropy/blob/master/setup.cfg#L144

Reply:
maybe add back an if/else with a non version `HAS_MATPLOTLIB`?

Reply:
(otherwise, we could use a pytest.importskip, but it's not a runtime any more)

Reply:
I added something to setup.cfg, let's hope that works!